### ISSUE TITLE
[Bug] GetFullPath returns unexpected results

### ISSUE THREAD

**Issue by marsej** [beefytech/Beef#968]
Hi everyone, I am very excited to find this language & ide and want to thank beefytech for taking time to design and create it. I'm still new to this but have started learning and programing in C# previously around 2003 but not continuously, more as a side hobby.

I attempted to make the following C# code in Beef and getting different results.

C#
 ```cs
 var di = new System.IO.DirectoryInfo(@".").EnumerateFileSystemInfos(@"*");
 foreach (var d in di) Console.WriteLine(d.FullName); //spit out fullpath+name (.Fullname) of every file and folder in build folder
 ```

Beef code:

			var fe = System.IO.Directory.Enumerate("*", .Directories | .Files);
			for (var f in fe)
			{
				var fp = scope String();
				f.GetFilePath(fp);
				var fpp = scope String();
				System.IO.Path.GetFullPath(fp, fpp);
				Console.WriteLine(fpp);
			}

Results from Beef code:

c:\BeefProj.toml
c:\BeefSpace.toml
c:\BeefSpace_User.toml
c:\build
c:\recovery

I don't have those files in c:\ so this is a surprising result. 0.43.0 Win 10

**Comment by marsej** [comment]
I found the problem to be the added \ by GetFilePath. GetFileName doesn't append that but ... 

```cs
			var fe = System.IO.Directory.Enumerate("*", .Directories | .Files);
			for (var f in fe)
			{
				var fp = scope String();
				f.GetFileName(fp);
				var fpp = scope String();
				System.IO.Path.GetFullPath(fp, fpp);
				Console.WriteLine(fpp);
			}
```

While that appears to work, it only does it for current folder. I'm not sure how to get this next example to work in Beef.
C#
```cs
var di = new System.IO.DirectoryInfo(@"..\..\..\").EnumerateFileSystemInfos(@"*");
            foreach (var d in di) Console.WriteLine(d.FullName);
```
Beef
```cs
			var fe = System.IO.Directory.Enumerate("..\\..\\..\\", .Directories | .Files);
			for (var f in fe)
			{
				var fp = scope String();
				//f.GetFilePath(fp);//Adde
				f.GetFileName(fp);
				var fpp = scope String();
				System.IO.Path.GetFullPath(fp, fpp);
				Console.WriteLine(fpp);
			}
```
The above example prints nothing.

**Comment by bfiete** [comment]
Fixed at 8c79e26d404d7320821e270e923cd50fcac4a401
